src/nvim/syntax.c

```diff
@@ -430,7 +430,7 @@ void syntax_start(win_T *wp, linenr_T lnum)
     } else {
       // Need to parse "minlines" lines before state can be considered
       // valid to store.
-      first_stored = current_lnum + syn_block->b_syn_sync_minlines;
+      first_stored = current_lnum + (linenr_T)syn_block->b_syn_sync_minlines;
```

Comment:
Will it make more sense to check the three `long`s in `synblock_T` to `int32_t`?

Reply:
Sounds good.